File: ru.kmorozov.library.data.client/src/main/java/ru/kmorozov/library/data/client/WebSocketEventHandler.java
package ru.kmorozov.library.data.client;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.messaging.simp.SimpMessagingTemplate;
import org.springframework.stereotype.Component;

/**
 * Created by sbt-morozov-kv on 04.05.2017.
 */

@Component
public class WebSocketEventHandler {

    private final SimpMessagingTemplate websocket;

    @Autowired
    public WebSocketEventHandler(SimpMessagingTemplate websocket) {
        this.websocket = websocket;
    }

    public void sendInfo(String message) {
        this.websocket.convertAndSend(WebSocketConfiguration.MESSAGE_PREFIX + "/info", message);
    }
}

File: ru.kmorozov.library.data.client/src/main/java/ru/kmorozov/library/data/client/WebSocketEventHandler.kt
package ru.kmorozov.library.data.client

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.messaging.simp.SimpMessagingTemplate
import org.springframework.stereotype.Component

/**
 * Created by sbt-morozov-kv on 04.05.2017.
 */

@Component
class WebSocketEventHandler @Autowired
constructor(private val websocket: SimpMessagingTemplate) {

    fun sendInfo(message: String) {
        this.websocket.convertAndSend(WebSocketConfiguration.MESSAGE_PREFIX + "/info", message)
    }
}
